mkosi/resources/__init__.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: PSF-2.0
+# Based on code from https://github.com/python/cpython/blob/main/Lib/importlib/resources/_common.py
+
+import contextlib
+import functools
+import os
+import stat
+import tempfile
+from collections.abc import Iterator
+from contextlib import AbstractContextManager
+from importlib.abc import Traversable
+from pathlib import Path
+
+
+def make_executable_if_script(path: Path) -> Path:
+    with path.open() as f:
+        buf = f.read(3)
+        if buf == "#!/":
+            os.chmod(path, path.stat().st_mode | stat.S_IEXEC)
+
+    return path
+
+
+@contextlib.contextmanager
+def temporary_file(path: Traversable, suffix: str = "") -> Iterator[Path]:
+    # Not using tempfile.NamedTemporaryFile as it leads to deeper 'try'
+    # blocks due to the need to close the temporary file to work on Windows
+    # properly.
+    fd, raw_path = tempfile.mkstemp(suffix=suffix)
```

Comment:
Since this moves away from the original, this comment can probably go, as this will never be used in Windows now.